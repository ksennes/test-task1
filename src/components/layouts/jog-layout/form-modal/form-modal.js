import React from "react";
import { format, parse} from "date-fns";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

import "./form-modal.scss";

import cancel from "../../../../img/cancel.png";

export const FormModal = ({ show, onHide, onSubmit, jogData, handleChange }) => {
  const token = localStorage.getItem("token");

  return (
    <div
      className="modal"
      style={show ? { display: "block" } : { display: "none" }}
    >
      <div className="cancel" onClick={() => onHide()}>
        <img src={cancel} />
      </div>
      <form className="form" onSubmit={e => {
          e.preventDefault();
          onSubmit(jogData, token);
        }}>
        <div className="form-group">
          <label>Distance</label>
          <input
            type="text"
            className="form-control"
            id="distance"
            value={jogData.distance || ''}
            onChange={(e) => handleChange(e)}
          />
        </div>
        <div className="form-group">
          <label>Time</label>
          <input
            type="text"
            className="form-control"
            id="time"
            value={jogData.time || ''}
            onChange={(e) => handleChange(e)}
          />
        </div>
        <div className="form-group">
          <label>Date</label>
          <DatePicker
            selected={parse(jogData.date, 'dd.MM.yyyy', new Date())}
            onChange={date =>
              handleChange({
                target: {
                  id: "date",
                  value: format(date, 'dd.MM.yyyy'),
                },
              })
            }
            dateFormat="dd.MM.yyyy"
            className='form-control'
          />
        </div>
        <button type="submit" className="submit-button" >
          Save
        </button>
      </form>
    </div>
  );
};
