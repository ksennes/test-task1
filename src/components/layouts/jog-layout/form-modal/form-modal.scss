.modal {
  position: fixed;
  top: 20%;
  height: 370px;
  border-radius: 44px;
  background-color: #7ed321;
  padding: 80px 88px 47px 88px;

  .form {
    width: 320px;

    .submit-button {
      width: 320px;
      height: 42px;
      border-radius: 25.2px;
      border: solid 2px #fff;
      background-color: #7ed321;
      margin-top: 20px;
      font-size: 12.6px;
      font-weight: bold;
      color: #fff;
    }
  }

  .cancel {
    position: absolute;
    top: 27px;
    right: 27px;
  }

  .cancel:hover {
    cursor: pointer;
  }

  .form-group {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 25px;

    label {
      font-size: 16px;
    }

    .form-control {
      padding: 0 10px;
      width: 236px;
      height: 31px;
      border-radius: 7px;
      border: solid 1px #979797;
      background-color: #fff;
    }
  }
}


@media (max-width: 750px) {
  .modal {
    height: 380px;
    padding: 85px 35px 37px 35px;

    .cancel {
      img {
        max-width: 18px;
      }
    }
  }
}