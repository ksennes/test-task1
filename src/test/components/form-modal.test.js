import React from 'react';
import { FormModal } from '../../components/layouts/jog-layout/form-modal/form-modal';
import renderer from 'react-test-renderer';

describe('FormModal Component', () => {
  it('sould match the last snapshot', () => {
      const handler = jest.fn();
      const jog = {
          distance: '15', 
          time: '15', 
          date: '20.02.2000'}
    const props = {
        show: false, 
        onHide: handler, 
        onSubmit: handler, 
        jogData: jog, 
        handleChange: handler
    }
    const wrapper = renderer.create(<FormModal {...props}/>)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});