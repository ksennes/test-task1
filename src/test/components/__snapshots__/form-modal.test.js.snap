// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormModal Component sould match the last snapshot with props 1`] = `
<div
  className="modal"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="cancel"
    onClick={[Function]}
  >
    <img
      src="test-file-stub"
    />
  </div>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        Distance
      </label>
      <input
        className="form-control"
        id="distance"
        onChange={[Function]}
        type="text"
        value="15"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Time
      </label>
      <input
        className="form-control"
        id="time"
        onChange={[Function]}
        type="text"
        value="15"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Date
      </label>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            className="form-control"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={false}
            type="text"
            value="20.02.2000"
          />
        </div>
      </div>
    </div>
    <button
      className="submit-button"
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;
